### PR TITLE
MySQL Tasks Elizaveta Panasiuk

### PR DESCRIPTION
**Pull request by ElizavetaPanasiuk**
https://travis-ci.com/github/ElizavetaPanasiuk/db-assignments

### CODE REVIEW

File: task/sql-tasks.js [hunk 1]
```diff
@@ -177,7 +272,13 @@ async function task_1_12(db) {
  *
  */
 async function task_1_13(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+        SELECT 
+            COUNT(ProductID) AS "TotalOfCurrentProducts",
+            SUM(Discontinued) AS "TotalOfDiscontinuedProducts"
```
Comment (AlexMokin) [inline task/sql-tasks.js]: use count, it shows intent better
even better just write 2 subqueries and a select without from-clause

eg

```
select 
 (select count(*) ....) as 'T'
 (select count(*) ....) as 'TT'
```

File: task/sql-tasks.js [hunk 2]
```diff
@@ -199,7 +308,24 @@ async function task_1_14(db) {
  *
  */
 async function task_1_15(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+        SELECT 
+            SUM(MONTH(OrderDate) = 1) AS "January",
```
Comment (AlexMokin) [inline task/sql-tasks.js]: use count to show intent explicitly

File: task/sql-tasks.js [hunk 3]
```diff
@@ -280,7 +463,25 @@ async function task_1_21(db) {
...
+        OrderDetails.UnitPrice AS "PricePerItem"
+    FROM Customers
+    INNER JOIN Orders ON Customers.CustomerID = Orders.CustomerID 
+    INNER JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID
+    INNER JOIN Products ON OrderDetails.ProductID = Products.ProductID
+    WHERE OrderDetails.UnitPrice = (SELECT MAX(OrderDetails2.UnitPrice)
+        FROM Customers AS Customers2
+        INNER JOIN Orders AS Orders2 ON Customers2.CustomerID = Orders2.CustomerID 
+        INNER JOIN OrderDetails AS OrderDetails2 ON Orders2.OrderID = OrderDetails2.OrderID 
+        WHERE Customers2.CompanyName = Customers.CompanyName)
```
Comment (AlexMokin) [inline task/sql-tasks.js]: companyname doesn't uniquely identify customer according to the schema. Also this column doesn't have an index and it makes the query slow

**Comment by AlexMokin** [comment]
Looks good. See comments.
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Review by AlexMokin — COMMENTED** [review]
Looks good. See comments.
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Review by ElizavetaPanasiuk — COMMENTED** [review]
https://travis-ci.com/github/ElizavetaPanasiuk/db-assignments/builds/182588511

**Comment by AlexMokin** [comment]
accepted

**Pull request closed without merging**